src/oc/hoc_oop.cpp

```diff
@@ -1497,24 +1500,24 @@ void hoc_endtemplate(Symbol* t) {
 	if (strcmp(ts->name, t->name) != 0) {
 		hoc_execerror(t->name, "- end template mismatched with begin");
 	}
-	ts->u.template->dataspace_size = icntobjectdata;
-	ts->u.template->symtable = hoc_symlist;
-	ts->u.template->count = 0;
-	ts->u.template->index = 0;
-	ts->u.template->olist = hoc_l_newlist();
-	ts->u.template->observers = (void*)0;
+	ts->u.ctemplate->dataspace_size = icntobjectdata;
+	ts->u.ctemplate->symtable = hoc_symlist;
+	ts->u.ctemplate->count = 0;
+	ts->u.ctemplate->index = 0;
+	ts->u.ctemplate->olist = hoc_l_newlist();
+	ts->u.ctemplate->observers = nullptr;
 	hoc_symlist = (poptemplate())->symlist;
-	free_objectdata(hoc_objectdata, ts->u.template);
+	free_objectdata(hoc_objectdata, ts->u.ctemplate);
 	hoc_thisobject = (poptemplate())->o;
 	hoc_in_template = (poptemplate())->i;
 	hoc_objectdata = (poptemplate())->odata;
 	icntobjectdata =  (poptemplate())->i;
-ts->u.template->init = s = hoc_table_lookup("init", ts->u.template->symtable);
+ts->u.ctemplate->init = s = hoc_table_lookup("init", ts->u.ctemplate->symtable);
 	if (s && s->type != PROCEDURE) {
 hoc_execerror("'init' can only be used as the initialization procedure for new objects",
 				(char *)0);
```

Comment:
```suggestion
				nullptr);
```
this should be applied in many other places too, of course.